### PR TITLE
Create guide section and add general paradigms topic

### PR DESCRIPTION
**Pull request by hamishwillee**
@julianoes This adds the guide section and first guide topic. The first topic is supposed to include everything the main "general" knowledge you have to know to use the API - error handling, how to create the library etc. 

Not sure where instructions for the CMakeLists.txt file should live. Possibly here, but more likely in a separate doc.

### CODE REVIEW

File: en/guide/general_usage.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+All other objects are returned/owned by the API and will automatically be destroyed along with `DroneCore`.
+
+
+## Error Handling
+
+DroneCore APIs do not raise exceptions! Instead, methods that can fail return success or an error reason as `enum` values.
+
+> **Tip** The error code usually reflects acknowledgment from the vehicle that it will perform the requested action (or not). The operation itself may not yet have completed (e.g. taking off).
+
+The various classes also all provide methods getting human readable strings from their associated enum (e.g. [DroneCore::connection_result_str()](h../api_reference/classdronecore_1_1_drone_core.md#classdronecore_1_1_drone_core_1a84c40dcefcafe888c38a5ed8dd93b0af), [Telemetry::result_str()](../api_reference/classdronecore_1_1_telemetry.md#classdronecore_1_1_telemetry_1a05c6355b7f8743250b2a7a611ea5fb4a)). You can see how these are used in the example code.
```
Comment (julianoes) [inline en/guide/general_usage.md]: Is this link correct `h../api_reference/classdronecore_...`? What is this `h`?

File: en/guide/general_usage.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+DroneCore has been designed to manage *aircraft* (VTOL, copter, fixed wing) that use the PX4 autopilot. 
+
+The APIs include methods that do not make sense for other vehicle types - including "takeoff" and "land". While ground vehicles may work, they are not supported and are untested.
+
+
+### Telemetry/Information
+
+DroneCore `Telemetry` information may not precisely reflect the state of the connected vehicle.
+
+DroneCore gets and stores vehicle state information from received MAVLink messages. The API is likely to more accurately reflect the device state if DroneCore is queried just after receiving an update rather than just before getting the next one.
```
Comment (julianoes) [inline en/guide/general_usage.md]: If you subscribe to the callbacks though, you almost get the latest info immediately.

File: en/guide/general_usage.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+So, for example, the [Action::land()](../api_reference/classdronecore_1_1_action.md#classdronecore_1_1_action_1ad1a50dd7bff99d3099916576efbf8cf6) method returns as soon as the vehicle confirms it will land, but will actually land at some later point. 
+
+The implication is that developers will need to separately monitor the completion of the requested actions, if this is important to the application.
+
+
+### Missions
+
+The `Mission` and `MissionItem` APIs provide a the most useful *subset* of MAVLink mission commands as a developer-friendly API. 
+
+Not every mission command behaviour supported by the protocol and PX4 will be supported by DroneCore. For example, at time of writing the API does not allow you to specify commands that jump back to previous commands.
```
Comment (julianoes) [inline en/guide/general_usage.md]: I would say `specify commands that jump back to previous mission items`.

File: en/guide/general_usage.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+### Connection Status
+
+Connection state information may be up to 3 seconds out of date (callbacks are notified 3 seconds after the last heartbeat message).
```
Comment (julianoes) [inline en/guide/general_usage.md]: Well, the connection state is still ok. It is only considered "timed out" after 3 seconds... 😄

File: en/guide/general_usage.md (hunk 1 above)
Comment (hamishwillee) [reply en/guide/general_usage.md]: No, will fix.

File: en/guide/general_usage.md (hunk 2 above)
Comment (hamishwillee) [reply en/guide/general_usage.md]: Good point. Updated.

File: en/guide/general_usage.md (hunk 3 above)
Comment (hamishwillee) [reply en/guide/general_usage.md]: Thanks. Fixed

File: en/guide/general_usage.md (hunk 4 above)
Comment (hamishwillee) [reply en/guide/general_usage.md]: Yoiks. Fixed.

**Comment by hamishwillee** [comment]
Hi @julianoes ,

Thanks for the review. I have fixed the points and merged. Can you please quickly scan and confirm that you're happy with the changes.

**Comment by julianoes** [comment]
Ok, all good.

**Pull request merged**